staging/src/k8s.io/client-go/tools/clientcmd/overrides.go

```diff
@@ -145,6 +145,7 @@ const (
 	FlagContext          = "context"
 	FlagNamespace        = "namespace"
 	FlagAPIServer        = "server"
+	FlagAPIServerName    = "server-name"
```

Comment:
I expect this will be confused with `--server`. Should we call this something like `--tls-server-name` or `--tls-verify-server-name` to make it clearer this is *only* used for TLS handshake verification?

Reply:
I also expected to see this in RecommendedClusterOverrideFlags / BindClusterFlags

Reply:
+1 to naming request, I think "TLSServerName" is clear enough

...We could maybe use this to not need so many SANs, that could be pretty nice.

Reply:
Renamed it to `--tls-server-name`